include/experimental/__p2630_bits/submdspan_mapping.hpp

```diff
@@ -322,6 +402,21 @@ struct deduce_layout_right_submapping<
   }
 };
 
+// We are reusing the same thing for layout_right and layout_right_padded
+// For layout_right as source StaticStride is static_extent(Rank-1)
+template<class Extents, size_t NumGaps, size_t StaticStride>
+struct Compute_S_static_layout_right {
+  // Neither StaticStride nor any of the looked for extents can zero.
+  // StaticStride never can be zero, the static_extents we are looking at are associated with
+  // integral slice specifiers - which wouldn't be valid for zero extent
+  template<size_t ... Idx>
+  MDSPAN_INLINE_FUNCTION
+  static constexpr size_t value(std::index_sequence<Idx...>) {
+    size_t val = ((Idx >= Extents::rank() - 1 - NumGaps && Idx < Extents::rank() - 1 ? (Extents::static_extent(Idx) == dynamic_extent?0:Extents::static_extent(Idx)) : 1) * ... * (StaticStride == dynamic_extent?0:StaticStride));
+    return val == 0?dynamic_extent:val;
```

Comment:
```suggestion
    if (StaticStride == dynamic_extent)
      return dynamic_extent;
    size_t val = StaticStride;
    for (size_t idx : {Idx...})
      if (idx >= Extents::rank() - 1 - NumGaps && idx < Extents::rank() - 1) {
        if (Extents::static_extent(idx) == dynamic_extent)
          return dynamic_extent;
        else
          val *= Extents::static_extent(idx);
      }
    return val;
```
Is there a guarantee that `Extents::rank()>0` and `Extents::rank() >= NumGaps + 1`?

Reply:
Yeah we are short circuiting Extents::rank()==0 (i.e. just return *this) and if NUmGaps == Extents::rank() we also short circuit because that is SubExtents::Rank() == 0

Reply:
Also: see above comment for compile time impact.